lighthouse-core/scripts/pptr-run-devtools.js

```diff
@@ -105,6 +112,13 @@ const startLighthouse = `
   const panel = UI.panels.lighthouse || UI.panels.audits;
   const button = panel.contentElement.querySelector('button');
   if (button.disabled) throw new Error('Start button disabled');
+
+  UI.dockController.setDockSide('undocked');
+
+  // Give the main target model a moment to be available.
+  // Otherwise, 'SDK.TargetManager.TargetManager.instance().mainTarget()' is null.
+  await new Promise(resolve => setTimeout(resolve, 1000));
```

Comment:
Could we create a [target observer](https://source.chromium.org/chromium/chromium/src/+/main:third_party/devtools-frontend/src/front_end/core/sdk/TargetManager.ts;l=155-163?q=targetmanager&ss=chromium%2Fchromium%2Fsrc:third_party%2Fdevtools-frontend%2Fsrc%2F) for this instead? I'd like to avoid waiting for arbitrary amounts of time if possible.